### PR TITLE
Move BMA and webadmin paths to satisfy the CI, full support to run with duniter user, upgrade support

### PR DESCRIPTION
**Pull request by M5oul**
- Precise progression message
- Update French description
- Remove `functions.sh` now useless
- Add upgrade support datadir move
- Create `duniter` user with `--home-dir` as `\$datadir`
- Protect webadmin, mv BMA and webadmin paths to make the CI happy
- Disclaimer: How to run Duniter commands with `duniter` user
- Update doc about new paths for BMA and webadmin
- Bump package to `1.8.2~ynh1`

### CODE REVIEW

**Comment by M5oul** [comment]
!testme

**Comment by M5oul** [comment]
!testme

**Comment by M5oul** [comment]
!testme

**Comment by M5oul** [comment]
!testme

File: scripts/install [hunk 1]
```diff
@@ -59,25 +59,26 @@ ynh_app_setting_set --app=$app --key=port --value=$port
...
+ynh_system_user_create --username=$app --use_shell
 
 #=================================================
 # NGINX CONFIGURATION
 #=================================================
-ynh_script_progression --message="Configuring NGINX web server..."
+ynh_script_progression --message="Configuring NGINX web server…"
 
 # Create a dedicated NGINX config
+sed -i "s@__PORT__@$port@g" ../conf/nginx.conf
```
Comment (yalh76) [inline scripts/install:73]: should not be needed, as `ynh_add_nginx_config` use now `ynh_add_config` https://yunohost.org/fr/packaging_apps_helpers?q=%2Fpackaging_apps_helpers#ynh_add_config

File: scripts/install [hunk 2]
```diff
@@ -101,46 +102,50 @@ chown -R $app:www-data "$datadir"
...
+duniter --home $datadir config --ws2p-host 127.0.0.1 --ws2p-port 20901 --ws2p-remote-host $domain --ws2p-remote-port 443 --ws2p-remote-path "/ws2p" --ws2p-noupnp
 
 #=================================================
 # SETUP SYSTEMD
 #=================================================
-ynh_script_progression --message="Configuring a systemd service..."
+ynh_script_progression --message="Configuring a systemd service…"
 
 # Create a dedicated systemd config
+sed -i "s@__DATADIR__@$datadir@g" ../conf/systemd.service
```
Comment (yalh76) [inline scripts/install:116]: should not be needed, as `ynh_add_systemd_config` use now `ynh_add_config` https://yunohost.org/fr/packaging_apps_helpers?q=%2Fpackaging_apps_helpers#ynh_add_config

File: scripts/upgrade [hunk 3]
```diff
@@ -69,25 +69,26 @@ fi
...
+ynh_system_user_create --username=$app --use_shell
 
 #=================================================
 # NGINX CONFIGURATION
 #=================================================
-ynh_script_progression --message="Upgrading NGINX web server configuration..."
+ynh_script_progression --message="Upgrading NGINX web server configuration…"
 
 # Create a dedicated NGINX config
+sed -i "s@__PORT__@$port@g" ../conf/nginx.conf
```
Comment (yalh76) [inline scripts/upgrade:83]: should not be needed

File: scripts/upgrade [hunk 4]
```diff
@@ -96,40 +97,66 @@ tempdir="$(mktemp -d)"
...
 
+#=================================================
+# MOVE DUNITER DIRECTORY
+#=================================================
+datadir=/home/yunohost.app/$app
+if [[ ! -d "${datadir}" ]]; then
+	ynh_script_progression --message="Moving data directory…"
+	mv /root/.config/duniter $datadir
+	chown duniter: $datadir
+fi
```
Comment (yalh76) [inline scripts/upgrade:108]: Maybe you should manage the `chmod` and `chown` on datadir after creating it

File: scripts/upgrade [hunk 5]
```diff
@@ -96,40 +97,66 @@ tempdir="$(mktemp -d)"
...
+duniter --home $datadir config --ws2p-host 127.0.0.1 --ws2p-port 20901 --ws2p-remote-host $domain --ws2p-remote-port 443 --ws2p-remote-path "/ws2p" --ws2p-noupnp
 
 #=================================================
 # SETUP SYSTEMD
 #=================================================
-ynh_script_progression --message="Upgrading systemd configuration..."
+ynh_script_progression --message="Upgrading systemd configuration…"
 
 # Create a dedicated systemd config
+sed -i "s@__DATADIR__@$datadir@g" ../conf/systemd.service
```
Comment (yalh76) [inline scripts/upgrade:124]: should not be needed

File: scripts/install (hunk 1 above)
Comment (M5oul) [reply scripts/install]: Thanks for the info! Done.

File: scripts/upgrade (hunk 4 above)
Comment (M5oul) [reply scripts/upgrade]: Done.

**Comment by M5oul** [comment]
!testme

**Comment by M5oul** [comment]
testme!

**Comment by M5oul** [comment]
!testme

**Comment by M5oul** [comment]
!testme

**Comment by M5oul** [comment]
!testme

**Comment by rodinux** [comment]
:crossed_fingers:

**Comment by rodinux** [comment]
hello, I have try an upgrade from this branch, but it fails. here's the logs: https://paste.yunohost.org/raw/zoniriqoso

**Comment by M5oul** [comment]
!testme

**Comment by M5oul** [comment]
Tests passed!
YunoHost-Apps team, please do not merge!
I am the maintainer of this package in the first place.

**Comment by M5oul** [comment]
!testme

**Pull request merged**